decision-records/006-javascript-compatibility.md

```diff
@@ -0,0 +1,161 @@
+# JavaScript browser compatibility
+
+Date: 27/06/23
+
+Status: Accepted
+
+## Decision
+
+We will ship JavaScript that can parse without errors in browsers supporting `<script type="module">`.
+
+We will update our guidance to tell service teams to include the JavaScript from GOV.UK Frontend using `<script type="module">`.
+
+We will set up:
+
+- transpiling to transform modern JavaScript to ES2015
+- linting to prevent us from using newer features and syntaxes unless we have actively opted in to them
+
+Going forward, when we want to use a newer feature we will disable the linting rule only after checking the performance impact of the transformation.
+
+Our current components do not require any polyfills. We set up the transpilation to only make [syntax transformations](https://babeljs.io/docs/babel-preset-env) and [bundle (to avoid having a dependency) Babel helpers](https://github.com/rollup/plugins/tree/master/packages/babel#babelhelpers) , not automatically [inject third party polyfills](https://github.com/babel/babel-polyfills). We will revisit this when we want to introduce a new feature that would require polyfills.
```

Comment:
```suggestion
Our current components do not require any polyfills. We set up the transpilation to only make [syntax transformations](https://babeljs.io/docs/babel-preset-env) and [bundle (to avoid having a dependency) Babel helpers](https://github.com/rollup/plugins/tree/master/packages/babel#babelhelpers), not automatically [inject third party polyfills](https://github.com/babel/babel-polyfills). We will revisit this when we want to introduce a new feature that would require polyfills.
```